src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+
+      if (herbivore.health < 1) {
```

Comment:
Since this is a herbivore's behavior it would make sense to allow the `Herbivore` class handle it and not move into the `Carnivore` class, could you come up with a way to achieve this?

Tip: remember `getters` and `setters`

**This is a personal suggestion/challenge, doing it this way is not required**